spec/beaker/shared/semvar_spec.rb

```diff
@@ -43,6 +43,180 @@ module Shared
+          expect( subject.puppet_version_split(ver).build ).to be == 8
+          expect( subject.puppet_version_split(ver).sha ).to be == 'f80879a'
+        end
+
+        it 'accepts valid PE final builds' do
+          ver = '2015.3.0'
+          expect( subject.puppet_version_split(ver).x ).to be == 2015
+          expect( subject.puppet_version_split(ver).y ).to be == 3
+          expect( subject.puppet_version_split(ver).z ).to be == 0
+          expect( subject.puppet_version_split(ver).rc ).to be == 999
+          expect( subject.puppet_version_split(ver).build ).to be == 0
+          expect( subject.puppet_version_split(ver).sha ).to be == 'none'
+        end
+
+        it 'accepts valid PE post-final builds' do
+          ver = '2015.3.0-8-gf80879a'
+          expect( subject.puppet_version_split(ver).x ).to be == 2015
+          expect( subject.puppet_version_split(ver).y ).to be == 3
+          expect( subject.puppet_version_split(ver).z ).to be == 0
+          expect( subject.puppet_version_split(ver).rc ).to be == 999
+          expect( subject.puppet_version_split(ver).build ).to be == 8
+          expect( subject.puppet_version_split(ver).sha ).to be == 'f80879a'
+        end
+
+        it 'accepts valid PE year/release' do
+          ver = '2015.3'
+          expect( subject.puppet_version_split(ver).x ).to be == 2015
+          expect( subject.puppet_version_split(ver).y ).to be == 3
+          expect( subject.puppet_version_split(ver).z ).to be == 0
+          expect( subject.puppet_version_split(ver).rc ).to be == 999
+          expect( subject.puppet_version_split(ver).build ).to be == 0
+          expect( subject.puppet_version_split(ver).sha ).to be == 'none'
+        end
+
+        it 'accepts valid PE year' do
+          ver = '2015'
+          expect( subject.puppet_version_split(ver).x ).to be == 2015
+          expect( subject.puppet_version_split(ver).y ).to be == 0
+          expect( subject.puppet_version_split(ver).z ).to be == 0
+          expect( subject.puppet_version_split(ver).rc ).to be == 999
+          expect( subject.puppet_version_split(ver).build ).to be == 0
+          expect( subject.puppet_version_split(ver).sha ).to be == 'none'
+        end
+
+        it 'rejects an invalid semver' do
+          expect { subject.puppet_version_split('2015.0-3') }.to raise_error(/Unknown version format/)
+        end
+
+        it 'rejects an invalid PE version' do
+          expect { subject.puppet_version_split('2015.1.0-rc5-300-null') }.to raise_error(/Unknown version format/)
+        end
+
+        it 'rejects a non version string' do
+          expect { subject.puppet_version_split('banana') }.to raise_error(/Unknown version format/)
+        end
+      end
+
+      describe 'puppet_version_comparison' do
+        it 'reports 2015.3.0-rc0-8-gf80879a is less than 2016' do
+          expect( subject.puppet_version_comparison( '2015.3.0-rc0-8-gf80879a', '<', '2016' ) ).to be === true
```

Comment:
@highb I'm curious about this calling style. Why don't we have our own `PuppetVersion` object that inherits from `Semantic::Version` that implements the Ruby version of what I think of as Java's Comparable interface, so that we can just use Ruby's comparison operators rather than passing in the comparison operator as another string parameter?